### DISCUSSION TITLE
docs: packageRules registryUrls

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#17158] [Suggest an Idea]
Wanted to check if the documentation for packageRules registryUrls is accurate, or needed up be updated / moved.

Currently the [registryUrls documentation](https://docs.renovatebot.com/configuration-options/#registryurls) appears under the Repository docs section.  The document here uses a docker datasource package rule with custom registryUrls.

When looking at the documentation for [packageRules](https://docs.renovatebot.com/configuration-options/#packagerules)  there is no mention of `registryUrls`.

Similarly, the published [renovate json schema](https://docs.renovatebot.com/renovate-schema.json) for IDE autocomplete matches the above (has registryUrls at the root of the configuration and not available under a packageRule)

Should both the json schema and documentation be updated to move `registryUrls` under `packageRules`? 🤔

**Comment by viceice** [comment]
No, you can also globally set registry urls. you can also use it in package rules or any other global object like `npm`, `python` or `nuget`

**Comment by rarkins** [comment]
In theory someone might nest registryUrls inside a manager or datasource block though, Eg

npm: {
  registryUrls
}

**Reply by viceice** [reply]
yep, i do it for my internal nuget sources 😉

**Comment by setchy** [comment]
> Should both the json schema and documentation be updated to move registryUrls under packageRules? 🤔

Rather than move, should `registryUrls` be added also to the packageRules json schema and documentation tree?

**Reply by viceice** [reply]
yes, the schema generation should be improved. as you can also use the undocumented `description` field in package rules

**Reply by setchy** [reply]
https://github.com/renovatebot/renovate/pull/17163 - feedback welcomed
